lib/space.rb

```diff
@@ -22,6 +24,11 @@ def self.add(name:, description:, price:, date:)
     Space.create(name: name, description: description, price: price, date: date)
   end
 
+  def book(name)
+    entry = Space.first(:name => name)
+    entry.update(:available => false)
+  end
+
 end
 
 DataMapper.finalize
```

Comment:
what does finalize do?

Reply:
It is called after all the models have been defined. It does a basic validity check of the DataMapper models